src/index/blockfilterindex.cpp

```diff
@@ -387,13 +389,25 @@ bool BlockFilterIndex::LookupFilter(const CBlockIndex* block_index, BlockFilter&
     return ReadFilterFromDisk(entry.pos, filter_out);
 }
 
-bool BlockFilterIndex::LookupFilterHeader(const CBlockIndex* block_index, uint256& header_out) const
+bool BlockFilterIndex::LookupFilterHeader(const CBlockIndex* block_index, uint256& header_out)
 {
+    LOCK(m_cs_headers_cache);
+    auto header = m_headers_cache.find(block_index->GetBlockHash());
+    if (header != m_headers_cache.end()) {
+        header_out = header->second;
+        return true;
+    }
+
     DBVal entry;
     if (!LookupOne(*m_db, block_index, entry)) {
         return false;
     }
 
+    // Add to the headers cache
+    if (m_headers_cache.size() < CF_HEADERS_CACHE_MAX_SZ) {
+        m_headers_cache.emplace(block_index->GetBlockHash(), entry.header);
+    }
+
```

Comment:
You'll still want to condition this on `block_index->nHeight % CFCHECKPT_INTERVAL == 0`, otherwise the most recent 2000 requested headers will be cached rather than the desired ones.

Reply:
yikes. Thanks

Reply:
done